app/src/features/apiClient/screens/apiClient/components/clientView/APIClientView.tsx

```diff
@@ -257,6 +258,10 @@ const APIClientView: React.FC<Props> = ({ apiEntry, apiEntryDetails, notifyApiRe
 
   const onSendButtonClick = useCallback(async () => {
     updateTab(apiEntryDetails?.id, { isPreview: false });
+    trackAPIRequestSent({
+      has_scripts: Boolean(entry.scripts?.preRequest?.length),
+      auth_type: entry?.auth?.currentAuthType,
+    });
```

Comment:
Request has not been sent yet. This is just the button click. After this if the request in found incorrect as per any validation check then the request wont be sent. Also if there's any error in prescript then also request won't be sent